drivers/flash/flash_shell.c

```diff
@@ -75,35 +87,47 @@ static int cmd_write(const struct shell *shell, size_t argc, char *argv[])
 	u32_t check_array[BUF_ARRAY_CNT];
 	u32_t buf_array[BUF_ARRAY_CNT];
 	struct device *flash_dev;
+	size_t fa_start;
 	u32_t w_addr;
+	char *endptr;
 	int j = 0;
 
-	flash_dev = device_get_binding(DT_FLASH_DEV_NAME);
-	if (!flash_dev) {
-		shell_error(shell, "Flash driver was not found!");
-		return -ENODEV;
-	}
-
 	if (argc < 2) {
 		shell_error(shell, "Missing address.");
 		return -EINVAL;
 	}
+	w_addr = strtoul(argv[1], &endptr, 16);
```

Comment:
i think that all this could be done in better way with less code duplication. Something like
```
args = (argc == 5) ? &argv[2] : &argv[1];
flash_dev = device_get_binding( (argc == 5) ? argv[1] : DT_FLASH_DEV_NAME);
if (flash_dev == NULL) shell_error(...)
w_addr = strtoul(&args[0], NULL, 16);
argc++; //to use later on to get data
```

Reply:
The `args = (argc == 5) ? ..' wouldn't work, since the number of arguments doesn't determine wether the first argument is a device or an address.  I can do `argv++;` to avoid the extra `fa_start`  variable.